DuckDuckGo.Fluent.Plugin/DuckDuckGoSearchApp.cs

```diff
@@ -0,0 +1,195 @@
+        }
+
+        public async ValueTask<ISearchResult> GetSearchResultForId(object searchObjectId)
+        {
+            DuckResult duckResult = searchObjectId switch
+            {
+                string json => JsonSerializer.Deserialize<DuckResult>(json),
+                DuckResult objectId => objectId,
+                _ => null
+            };
+
+            switch (duckResult)
+            {
+                case null:
+                    return default;
+                case { SearchResultType: ResultType.QrCode }:
+                {
+                    DuckDuckGoSearchResult duckGoSearchResult = await GetQrImage(duckResult.SearchedText);
+                    return duckGoSearchResult;
+                }
+            }
+
+            string url = GetEndpointUrl(duckResult.SearchedText);
+            DuckDuckGoApiResult apiResult = await HttpCalls.GetApiResult(url);
+
+            if (apiResult == null) return default;
+
+            var duckResultFactory = DuckResultFactory.Create(apiResult, duckResult.SearchedText);
+
+            DuckResult result = duckResult.SearchResultType switch
+            {
+                ResultType.Answer => duckResult, // Answers will often have numerical computations and are not suitable for custom tags as they often update very frequently.
+                ResultType.Definition => duckResultFactory.GetDefinition(),
+                ResultType.Abstract => duckResultFactory.GetAbstract(),
+                ResultType.QrCode => null,
+                ResultType.SearchResult => duckResult,
+                _ => null
+            };
+
+            return result == null ? default(ISearchResult) : GetISearchResult(result);
+        }
+
+        public ValueTask<IHandleResult> HandleSearchResult(ISearchResult searchResult)
+        {
+            if (searchResult is not DuckDuckGoSearchResult duckGoSearchResult)
+                throw new InvalidCastException(nameof(DuckDuckGoSearchResult));
+
+            string url = duckGoSearchResult.Url;
+
+            if (duckGoSearchResult.SelectedOperation is DuckDuckGoSearchOperations duckGoSearchOperations)
+            {
+                IProcessManager managerInstance = ProcessUtils.GetManagerInstance();
+                switch (duckGoSearchOperations.ActionType)
+                {
+                    case ActionType.OpenDuckDuckGo:
+                        if (!string.IsNullOrWhiteSpace(url))
+                            managerInstance.StartNewProcess(url);
+                        break;
+
+                    case ActionType.SaveImage:
+                        SaveQrImage(duckGoSearchResult);
+                        break;
+
+                    default:
+                        return default;
+                }
+            }
+            else if (duckGoSearchResult.SelectedOperation.OperationName == CopyUrl.OperationName)
+            {
+                if (!string.IsNullOrWhiteSpace(url))
+                    Clipboard.SetText(url);
```

Comment:
FYI - you can use FS built-in `CopySearchOperation` or `SelfRunCopySearchOperation` which will automatically copy the context and will not cause `HandleSearchResult` to invoke